patches/hhvm_exts.diff

```diff
@@ -46,32 +54,70 @@ index 6e94fa9..26a8a35 100644
 +	return (result);
 +}
 +
++/*
++ * Derived from:
++ * $OpenBSD: pctr.h,v 1.5 2014/03/29 18:09:28 guenther Exp$
++ *
++ * PHP has no way to represent a unsigned 64-bit integer. We return a signed
++ * 64. Any app-level code using this must know that this is really a unsigned
++ * value.
++ */
++int64_t HHVM_FUNCTION(read_ts_reg) {
++    u_int32_t hi, lo;
++
++    __asm volatile("rdtsc" : "=d" (hi), "=a" (lo));
++    return ((u_int64_t) hi << 32) | (u_int64_t) lo;
++}
++
++/*
++ * Print a TS register delta.
++ *
++ * We do it this way because PHP has no way to represent a unsigned 64.
++ */
++void HHVM_FUNCTION(print_ts_delta, const int64_t start, const int64_t stop) {
++    u_int64_t u_start = static_cast<u_int64_t>(start);
++    u_int64_t u_stop = static_cast<u_int64_t>(stop);
++
++    u_int64_t delta = u_stop - u_start;
++    std::cout << delta;
++}
++
  ///////////////////////////////////////////////////////////////////////////////
  }
- 
 diff --git a/hphp/runtime/ext/std/ext_std_misc.h b/hphp/runtime/ext/std/ext_std_misc.h
-index 5696eee..3b4ab87 100644
+index 49db8a5..865c480 100644
 --- a/hphp/runtime/ext/std/ext_std_misc.h
 +++ b/hphp/runtime/ext/std/ext_std_misc.h
-@@ -43,6 +43,7 @@ Variant HHVM_FUNCTION(unpack, const String& format, const String& data);
+@@ -48,6 +48,9 @@ Variant HHVM_FUNCTION(unpack, const String& format, const String& data);
  Array HHVM_FUNCTION(sys_getloadavg);
  Array HHVM_FUNCTION(token_get_all, const String& source);
  String HHVM_FUNCTION(token_name, int64_t token);
 +double HHVM_FUNCTION(clock_gettime_monotonic);
++int64_t HHVM_FUNCTION(read_ts_reg);
++void HHVM_FUNCTION(print_ts_delta, const int64_t start, const int64_t stop);
  String HHVM_FUNCTION(hphp_to_string, const Variant& v);
- extern const double k_INF;
- extern const double k_NAN;
+ Variant HHVM_FUNCTION(SystemLib_max2, const Variant& arg1, const Variant& arg2);
```

Comment:
Minor thing, but shouldn't it be `const Variant &arg1` (i.e. the ampersand next to the variable name)?